libraries/StorageManager/StorageManager.cpp

```diff
@@ -131,6 +136,19 @@ StorageAccess::StorageAccess(StorageManager::StorageType _type) :
 bool StorageAccess::read_block(void *data, uint16_t addr, size_t n) const
 {
     uint8_t *b = (uint8_t *)data;
+
+#if AP_SDCARD_STORAGE_ENABLED
+    if (file != nullptr) {
+        // using microSD data
+        if (addr > file->bufsize) {
+            return false;
```

Comment:
Callers such as `StorageAccess::read_byte`  will return uninitialised stack data if we take this return path.

Reply:
when we return false from a function we commonly don't fill in all the arguments
I can memset it I guess, but seems a waste

Reply:
the existing code behaves the same way

Reply:
> when we return false from a function we commonly don't fill in all the arguments I can memset it I guess, but seems a waste

Caller should work out what to do.  That's why the function isn't void.  Caller's fault in this case IMO.